### ISSUE TITLE
`va_start` type mismatch

### ISSUE THREAD

**Issue by Rua** [immunant/c2rust#1353]
When transpiling this C code

```C
void I_Error (char *error, ...)
{
    va_list argptr;
    va_start (argptr,error);
```

the following code is emitted
```rust
#[no_mangle]
pub unsafe extern "C" fn I_Error(mut error: *mut std::ffi::c_char, mut args: ...) {
    let mut argptr: va_list = 0 as *mut std::ffi::c_char;
    argptr = args.clone();
```

The last line creates a compile error, because `args` is a Rust type, `std::ffi::VaListImpl`, and doesn't match the type of `argptr`.

**Comment by fw-immunant** [comment]
varargs translation is dependent on a lot of implementation details. What platform are you running on, and what versions of c2rust and rustc are you using?

**Comment by Rua** [comment]
- NixOS 25.05, x86-64
- c2rust master from a few days ago
- Using the provided Nix flake for compiling c2rust
- Nightly 2022-08-07 for compiling the transpiled code

**Comment by Rua** [comment]
After investigating, it looks like this bit of code is meant to automatically replace variables declared with C's `va_list` type with Rust's `VaListImpl`:
https://github.com/immunant/c2rust/blob/master/c2rust-transpile/src/translator/mod.rs#L2682-L2696

But in my case, it seems that replacement is not triggered and the variable is left as `va_list`?

EDIT: In this program, the chain of typedefs appears to be translated as:
```
pub type va_list = __gnuc_va_list;
pub type __gnuc_va_list = __builtin_va_list;
pub type __builtin_va_list = *mut std::ffi::c_char;
```

The section of code I linked calls `TypedAstContext::is_va_list`, which uses `is_va_list_struct` to determine if the type is a `va_list`. But that function assumes that the type is a struct, while here it's a chain of typedefs that ends with `char*`.

From what I can tell, the function defined immediately above, named `is_builtin_va_list`, would find it by the `__builtin_va_list` typedef. But that function appears to be completely unused in the c2rust codebase.
